sound/soc/sof/ops.h

```diff
@@ -522,6 +522,15 @@ snd_sof_set_mach_params(struct snd_soc_acpi_mach *mach,
 		sof_ops(sdev)->set_mach_params(mach, sdev);
 }
 
+static inline struct snd_sof_of_mach *
+snd_sof_of_machine_select(struct snd_sof_dev *sdev)
+{
+	if (sof_ops(sdev) && sof_ops(sdev)->of_machine_select)
+		return sof_ops(sdev)->of_machine_select(sdev);
```

Comment:
I don't understand the definitions at all. In the DT world, the machine is  already described with a device, which will be matched with a driver already. What does 'of_machine_select' mean in this context?

Reply:
do you mean 'a machine is already described with a device using device tree, right?'. Device tree describes the hardware, while we need information like machine driver name, tplg name, firmware name which according to Mark cannot be placed inside a DT node. 

https://www.spinics.net/lists/devicetree/msg433913.html